src/data_access/api/SpotifyAPI.java

```diff
@@ -0,0 +1,109 @@
+package data_access.api;
+
+import entity.CommonSongFactory;
+import entity.Song;
+import entity.SongFactory;
+import io.github.cdimascio.dotenv.Dotenv;
+import okhttp3.*;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Random;
+
+public class SpotifyAPI implements SongAPI{
+    private final String CLIENT_ID;
+    private final String CLIENT_SECRET;
+    private final OkHttpClient client;
+    private String authToken = null;
+    private final SongFactory songFactory;
+
+    public SpotifyAPI(SongFactory songFactory){
+        Dotenv dotenv = Dotenv.configure().filename("api.env").load();
```

Comment:
Ill wrap in in a try-catch which is better practice for a situation like this. The api.env file is in the \src directory but doesn't seem to cause an issue. How about you @rayceramsay

Reply:
The original code works fine for me. @NickinAction is it possible you changed something with your build settings in Intellij or something?

Reply:
I don't think I've changed any build settings. I'm getting another error to do with an import statement in MultipleChoiceRound.java, that I didn't have before:

``` java
import org.checkerframework.checker.units.qual.A;
```

This might mean theres something wrong on my end. Also it seems that `git pull` isnt working properly for me either. I'll approve the PR now.